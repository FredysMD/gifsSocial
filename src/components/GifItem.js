import React from 'react';
import PropType from 'prop-types';

export const GifItem = ({ url, title }) => {

    return (
        <div className = "card-size">
             <div className = "card animate__slideInDown">
                <img className = "card-img-top" src = { url } alt = { title } />
                <div className = "card-body">
                    <h5 className = "card-title">{ title }</h5>
                </div>
            </div>
        </div>
    )
}

GifItem.prototype = {
    url: PropType.string.isRequired,
    title: PropType.string.isRequired
}