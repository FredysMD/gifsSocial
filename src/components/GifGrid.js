import React from 'react';
import Prototype from 'prop-types';

import { useGifFetch } from '../hooks/useGifFetch';

import { GifItem } from './GifItem';


const GifGrid = ( { category }) => {

    const { data:images, loading } = useGifFetch( category );

    return (
        <>
          <h2>{ category }</h2>
          { loading && <p className = "animate__animated animate__fadeIn">Loading...</p> }
          <ol>
            {
                images.map( img => (
                    <GifItem 
                        key = {img.id}
                        { ...img }
                    />
                ))
            }
          </ol>

        </>
    )
}

GifGrid.prototype = {
    category: Prototype.string.isRequired
}

export default GifGrid;

