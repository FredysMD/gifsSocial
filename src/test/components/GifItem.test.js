import React from 'react';
import { shallow } from 'enzyme';

import { GifItem } from '../../components/GifItem';


describe('Test to GifItem Component', () => {
    
    const props = { 
        url: 'https://urlTest.com', 
        title: 'Title test'
    };
    
    const wrapper = shallow( <GifItem title = { props.title } url = { props.url } /> );

    test('should have a component GifItem', () => {

        expect( wrapper ).toMatchSnapshot();
    
    });

    test('should have a title equal to "Title test" ', () => {
        
        const h5 = wrapper.find('h5').text().trim();

        expect( h5 ).toBe( props.title );

    });
    
    test('should have a image with src equal to "https://urlTest.com" ', () => {
        
        const imgUrl = wrapper.find('img').prop('src');

        expect( imgUrl ).toBe( props.url );

    });

    test('should have a image with alt equal to "Title test"', () => {
        
        const imgAlt = wrapper.find('img').prop('alt');

        expect( imgAlt ).toBe( props.title );

    });
    
    test('should have a div with className animate__slideInDown', () => {
        
        const animate = wrapper.find('div').at(1).prop('className');

        expect( animate.includes('animate__slideInDown') ).toBe( true );

    });
    
});
