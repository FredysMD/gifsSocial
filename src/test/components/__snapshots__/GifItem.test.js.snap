// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GifItem Component should have a component GifItem 1`] = `
<div
  className="card-size"
>
  <div
    className="card animate__slideInDown"
  >
    <img
      alt="Title test"
      className="card-img-top"
      src="https://urlTest.com"
    />
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        Title test
      </h5>
    </div>
  </div>
</div>
`;
